### PR TITLE
Add DECIMAL type

### PR DESCRIPTION
**Pull request by karteekmurthys**

### CODE REVIEW

File: velox/type/Decimal.h [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  operator StringView() const {VELOX_NYI()}
+
+  std::string toString() const;
+
+  operator std::string() const {
+    return toString();
+  }
+
+  bool operator==(const Decimal& other) const {
+    return true;
```
Comment (aditi-pandit) [inline velox/type/Decimal.h:83]: Use VELOX_NYI() for all the not implemented code.

File: velox/type/Variant.cpp [hunk 2]
```diff
@@ -517,6 +536,10 @@ uint64_t variant::hash() const {
       auto dateValue = value<TypeKind::DATE>();
       return folly::Hash{}(dateValue.days());
     }
+    case TypeKind::DECIMAL: {
+      auto decimalValue = value<TypeKind::DECIMAL>();
+      return folly::Hash{}(decimalValue.toString());
```
Comment (aditi-pandit) [inline velox/type/Variant.cpp:541]: This should use the internal integer representation of Decimal... Using String is too inefficient.

File: velox/type/Decimal.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+#include "velox/type/Decimal.h"
+
+namespace facebook::velox {
+void parseTo(folly::StringPiece in, Decimal& out) {
+  VELOX_NYI();
+}
+
+std::string Decimal::toString() const {
+  return "";
```
Comment (majetideepak) [inline velox/type/Decimal.cpp:25]: `VELOX_NYI()` ?

File: velox/type/Decimal.h [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+  uint8_t scale_ = kDefaultScale; // The number of digits on the right
+                                  // of radix point.
+};
+
+class DecimalCasts {
+ public:
+  static Decimal parseStringToDecimal(const std::string& value) {
+    // throws overflow exception if length is > 38
+    VELOX_CHECK_GT(
+        value.length(), 0, "Decimal string must have at least 1 char")
```
Comment (majetideepak) [inline velox/type/Decimal.h:154]: `1 character`

File: velox/type/Decimal.h [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+template <typename T>
+void toAppend(const ::facebook::velox::Decimal& value, T* result) {}
+} // namespace facebook::velox
+
+namespace std {
+template <>
+struct hash<::facebook::velox::Decimal> {
+  size_t operator()(const ::facebook::velox::Decimal& value) const {
+    return 0;
```
Comment (majetideepak) [inline velox/type/Decimal.h:223]: VELOX_NYI

File: velox/type/Decimal.h [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+};
+
+std::string to_string(const ::facebook::velox::Decimal& ts);
+} // namespace std
+
+namespace folly {
+template <>
+struct hasher<::facebook::velox::Decimal> {
+  size_t operator()(const ::facebook::velox::Decimal& value) const {
+    return 0;
```
Comment (majetideepak) [inline velox/type/Decimal.h:234]: VELOX_NYI

File: velox/type/tests/TypeTest.cpp [hunk 7]
```diff
@@ -198,6 +198,77 @@ TEST(Type, parseStringToDate) {
...
+  EXPECT_EQ(Decimal(-1400, 4, 4), DecimalCasts::parseStringToDecimal("-.1400"));
+  EXPECT_EQ(Decimal(1400, 4, 4), DecimalCasts::parseStringToDecimal("+.1400"));
+  EXPECT_EQ(
+      Decimal(14001234, 8, 4),
+      DecimalCasts::parseStringToDecimal("+1400.1234"));
+
+  EXPECT_EQ(
+      Decimal(123456789012345, 15, 5),
+      DecimalCasts::parseStringToDecimal(std::string("1234567890.12345")));
+  int128_t testValue = 0xAB54A98CEB1F0AD2;
```
Comment (majetideepak) [inline velox/type/tests/TypeTest.cpp:214]: testHexValue

**Comment by karteekmurthys** [comment]
Duplicate PR.

**Pull request closed without merging**